packages/core/src/table/table.scss

```diff
@@ -0,0 +1,158 @@
+  --background-shade: #{$cds-alias-object-container-background-shade};
+  --color: #{$cds-global-typography-color-400};
+  --border-radius: #{$cds-alias-object-border-radius-100};
+  --border-width: #{$cds-alias-object-border-width-100};
+  --header-separator-width: #{$cds-alias-object-border-width-100};
+  --border-color: #{$cds-alias-object-border-color};
+  --border-tint: #{$cds-alias-object-border-color-tint};
+  --cell-padding-vertical: #{$cds-global-space-5};
+  --cell-padding-horizontal: #{$cds-global-space-6};
+  --font-size: #{$cds-global-typography-body-font-size};
+}
+
+// Styles and default bordering
+[cds-table] {
+  background-color: var(--background);
+  border-collapse: separate;
+  border-spacing: 0;
+  color: var(--color);
+  margin: 0;
+  max-width: 100%;
+  width: 100%;
+
+  td,
+  th {
+    padding: var(--cell-padding-vertical) var(--cell-padding-horizontal);
+    text-align: center;
+    vertical-align: top;
+  }
+
+  th {
+    color: var(--color);
+    background-color: var(--background-shade);
+    border-bottom-width: var(--header-separator-width);
+    border-bottom-style: solid;
+    border-bottom-color: var(--border-color);
+    border-top-width: var(--border-width);
+    border-top-style: solid;
+    border-top-color: var(--border-color);
+  }
+
+  td {
+    background-color: var(--background);
+    border-bottom: var(--border-width) solid var(--border-tint);
+  }
+
+  tr th:first-child,
+  tr td:first-child {
+    border-left: var(--border-width) solid var(--border-color);
+  }
+
+  tr th:last-child,
+  tr td:last-child {
+    border-right: var(--border-width) solid var(--border-color);
+  }
+
+  tr:last-child td {
+    border-bottom-color: var(--border-color);
+  }
+
+  tfoot th {
+    border-top-width: 0;
+  }
+}
+
+// Border radius styles
+[cds-table] {
+  tr:first-child {
+    :first-child {
+      border-radius: 0;
+      border-top-left-radius: var(--border-radius);
+    }
+
+    :last-child {
+      border-radius: 0;
+      border-top-right-radius: var(--border-radius);
+    }
+  }
+
+  & > *:last-child > tr:last-of-type {
+    :first-child {
+      border-radius: 0;
+      border-bottom-left-radius: var(--border-radius);
+    }
+
+    :last-child {
+      border-radius: 0;
+      border-bottom-right-radius: var(--border-radius);
+    }
+  }
+}
+
+// Left Alignment options
+[cds-table~='left'] th,
+[cds-table~='left'] td,
+[cds-table] td[cds-col~='left'],
+[cds-table] th[cds-col~='left'] {
+  text-align: left;
+}
+
+// Right Alignment options
+[cds-table~='right'] th,
+[cds-table~='right'] td,
+[cds-table] td[cds-col~='right'],
+[cds-table] th[cds-col~='right'] {
+  text-align: right;
+}
+
+// No Border options
+[cds-table~='noborder'],
+[cds-table~='noborder'] tr th,
+[cds-table~='noborder'] tr td {
+  --border-width: 0;
+  background-color: inherit;
+}
+
+// Compact options
+[cds-table~='compact'] {
+  td,
+  th {
+    padding: calc(var(--cell-padding-vertical) / 2) calc(var(--cell-padding-horizontal) / 2);
```

Comment:
To preserve theming behavior, alternate style options through attrs should modify through the CSS props when possible.

```css
[cds-table~='compact'] {
  --cell-padding: calc(#{$cds-global-space-5} / 2) calc(#{$cds-global-space-6} / 2);
}
```

This allows users to define their own custom theme states by altering the custom properties based on the attr. An example of this, a user can adjust the look of cds-button in a particular state by hooking into the attr and custom prop.

```css
cds-button[status=success] {
  --background: purple;
}
```